starknet_py/net/models/transaction.py

```diff
@@ -1,28 +1,244 @@
+
+
+# pylint: disable=no-self-use, unused-argument
+@dataclass(frozen=True)
+class Transaction(ABC):
+    """
+    StarkNet transaction base class.
+    """
+
+    version: int = field(metadata={"marshmallow_field": Felt()})
+
+    @property
+    @abstractmethod
+    def tx_type(self) -> TransactionType:
+        """
+        Returns the corresponding TransactionType enum.
+        """
+
+    @abstractmethod
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network - a unique identifier of the
+        transaction. See :ref:`compute_transaction_hash` docstring for more details.
+        """
+
+
+@dataclass(frozen=True)
+class AccountTransaction(Transaction):
+    """
+    Represents a transaction in the StarkNet network that is originated from an action of an
+    account.
+    """
+
+    max_fee: int = field(metadata={"marshmallow_field": Felt()})
+    signature: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
+    )
+    nonce: int = field(metadata={"marshmallow_field": Felt()})
+
+
+@dataclass(frozen=True)
+class Declare(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is a declaration of a StarkNet contract
+    class.
+    """
+
+    contract_class: ContractClass
+    # The address of the account contract sending the declaration transaction.
+    sender_address: int = field(metadata={"marshmallow_field": Felt()})
+
+    @property
+    def tx_type(self) -> TransactionType:
+        return TransactionType.DECLARE
+
+    @marshmallow.post_dump
+    def compress_program_post_dump(
+        self, data: Dict[str, Any], many: bool, **kwargs
+    ) -> Dict[str, Any]:
+        data["type"] = "DECLARE"
+        return compress_program_post_dump(data=data, many=many)
+
+    @marshmallow.pre_load
+    def decompress_program(
+        self, data: Dict[str, Any], many: bool, **kwargs
+    ) -> Dict[str, Any]:
+        return decompress_program(data=data, many=many)
+
+    def calculate_hash(self, chain_id: StarknetChainId) -> int:
+        """
+        Calculates the transaction hash in the StarkNet network.
+        """
+        return compute_declare_transaction_hash(
+            contract_class=self.contract_class,
+            chain_id=chain_id.value,
+            sender_address=self.sender_address,
+            max_fee=self.max_fee,
+            version=self.version,
+            nonce=self.nonce,
+        )
+
+
+@dataclass(frozen=True)
+class DeployAccount(AccountTransaction):
+    """
+    Represents a transaction in the StarkNet network that is a deployment of a StarkNet account
+    contract.
+    """
+
+    class_hash: int = field(metadata={"marshmallow_field": Felt()})
+    contract_address_salt: int = field(metadata={"marshmallow_field": Felt()})
+    constructor_calldata: List[int] = field(
+        metadata={"marshmallow_field": fields.List(fields.String())}
```

Comment:
So calldata is a list of integers encoded as strings?

Reply:
Yes, this JSON of the `DeployAccount` transaction from the Cairo lang
```
{'contract_address_salt': '0x6ac08f1431abc887b9d28e5eb86c50f89752b27032718750fdc51d55d177b3d', 'constructor_calldata': ['160152079770464609983014864782684742393018887390785051974472089603158960695'], 'class_hash': '0x6989682284a6b09faca008c6b50bcf450723133cce14e10954c6f1b23613aa4', 'signature': ['3283987649632252841977199248871993511573663748602820833168218210980447668663', '229421868398535999618245223132720803182314358948341788119067290439773148182'], 'version': '0x1', 'max_fee': '0x56bc75e2d63100000', 'nonce': '0x0', 'type': 'DEPLOY_ACCOUNT'}
```